### PR TITLE
Fixes #37581 - Refactor timezone and NTP server configuration in kickstart

### PR DESCRIPTION
**Pull request by nofaralfasi**
- Split timezone and ntp-server configuration.
- Add conditional logic to use 'timesource' command for RHEL 9 and above, maintaining compatibility with older versions.

### CODE REVIEW

File: app/views/unattended/provisioning_templates/provision/kickstart_default.erb [hunk 1]
```diff
@@ -142,7 +142,15 @@ authselect --useshadow --passalgo=<%= @host.operatingsystem.password_hash.downca
 <% else -%>
 authconfig --useshadow --passalgo=<%= @host.operatingsystem.password_hash.downcase || 'sha256' %> --kickstart
 <% end -%>
-timezone --utc <%= host_param('time-zone') || 'UTC' %> <%= host_param('ntp-server') ? "--ntpservers #{host_param('ntp-server')}" : '' %>
+
+timezone --utc <%= host_param('time-zone') || 'UTC' %>
+<% if host_param('ntp-server') %>
+<% if rhel_compatible && os_major > 8 %>
```
Comment (ekohl) [inline app/views/unattended/provisioning_templates/provision/kickstart_default.erb]: I always prefer `>=` over `>` so I don't need to do mental math. Though in this particular case you can also revert it to `rhel_compatible && os_major < 9` and swap the branches. Then you gain non-EL support like Fedora.

**Review by ekohl — COMMENTED** [review]
Can you use the timezone command twice? I'd have assumed you can only use it once

File: app/views/unattended/provisioning_templates/provision/kickstart_default.erb (hunk 1 above)
Comment (nofaralfasi) [reply app/views/unattended/provisioning_templates/provision/kickstart_default.erb]: `timezone --ntpservers` is also deprecated in Oracle Linux 9 and Fedora 34. Do I need to check here all supported OSs?

**Comment by nofaralfasi** [comment]
> Can you use the timezone command twice? I'd have assumed you can only use it once

You're right. Thank you for noticing!

File: app/views/unattended/provisioning_templates/provision/kickstart_default.erb [hunk 2]
```diff
@@ -142,7 +142,15 @@ authselect --useshadow --passalgo=<%= @host.operatingsystem.password_hash.downca
 <% else -%>
 authconfig --useshadow --passalgo=<%= @host.operatingsystem.password_hash.downcase || 'sha256' %> --kickstart
 <% end -%>
+
+<% if rhel_compatible && os_major < 9 %>
```
Comment (ekohl) [inline app/views/unattended/provisioning_templates/provision/kickstart_default.erb]: Please use `-%>` to avoid empty lines

**Review by ekohl — CHANGES_REQUESTED** [review]
I think you need to update the snapshots

**Review by ekohl — CHANGES_REQUESTED** [review]
Test failures are caused by this. Note the trailing whitespace that's no longer there.

**Comment by stejskalleos** [comment]
@nofaralfasi you can update snapshots by running `RAILS_ENV=test bundle exec rails snapshots:generate`

**Comment by nofaralfasi** [comment]
> @nofaralfasi you can update snapshots by running `RAILS_ENV=test bundle exec rails snapshots:generate`

I wasn't aware of this option, and after many attempts, I updated it manually until all tests passed.

**Review by stejskalleos — DISMISSED** [review]
:green_apple: **LGTM**
Checked with the [official documentation](https://pykickstart.readthedocs.io/en/latest/kickstart-docs.html#timezone)

File: app/views/unattended/provisioning_templates/provision/kickstart_default.erb [hunk 3]
```diff
@@ -142,7 +142,15 @@ authselect --useshadow --passalgo=<%= @host.operatingsystem.password_hash.downca
 <% else -%>
 authconfig --useshadow --passalgo=<%= @host.operatingsystem.password_hash.downcase || 'sha256' %> --kickstart
 <% end -%>
+
```
Comment (ekohl) [inline app/views/unattended/provisioning_templates/provision/kickstart_default.erb]: I'd have a slight preference to keep this together with the other options. We already have too many nonsense empty lines in the rendered template.
```suggestion
```

**Review by ekohl — DISMISSED** [review]

**Review by stejskalleos — APPROVED** [review]

**Comment by stejskalleos** [comment]
Thanks @nofaralfasi @ekohl

**Pull request merged**